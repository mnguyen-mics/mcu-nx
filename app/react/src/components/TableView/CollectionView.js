import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Pagination, Spin } from 'antd';


function CollectionView(props) {

  const {
    collectionItems,
    gutter,
    span,
    pagination,
    loading,
  } = props;

  return loading ? (<Row style={{ height: '350px' }}><Row className="mcs-aligner"><Spin /></Row></Row>) : (
    <Row className="mcs-table-card">
      <Row gutter={gutter}>
        {collectionItems && collectionItems.map(item => {
          return (
            <Col span={span} key={item.id}>
              { item }
            </Col>
          );
        })}

      </Row>
      <Row className="text-right">
        <Pagination {...pagination} />
      </Row>
    </Row>
  );
}

CollectionView.defaultProps = {
  gutter: 20,
  span: 6,
  loading: false,
  collectionItems: null
};

CollectionView.propTypes = {
  collectionItems: PropTypes.arrayOf(PropTypes.element.isRequired),
  gutter: PropTypes.number,
  span: PropTypes.number,
  loading: PropTypes.bool,
  pagination: PropTypes.shape().isRequired,
};

export default CollectionView;
