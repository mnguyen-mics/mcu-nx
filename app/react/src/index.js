import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import configureStore from './store';
import { Navigator } from './containers/Navigator';

const store = configureStore();

const micsProvider = (
  <Provider store={store}>
    <Navigator />
  </Provider>
);

render(
  micsProvider,
  document.getElementById('mcs-react-app') // eslint-disable-line no-undef
);
