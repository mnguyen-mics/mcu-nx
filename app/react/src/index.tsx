/* eslint-disable */
import React from 'react';
import { render } from 'react-dom';

import App from './App';

declare global {
  interface Window {
    require: any;
    mainRev: any;
  }
}

render(<App />, document.getElementById('mcs-react-app'), () => {
  window.require([window.mainRev], () => {
    //
  });
});
