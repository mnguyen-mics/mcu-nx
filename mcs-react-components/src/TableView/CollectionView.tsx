import React from 'react';
import {Row, Col, Pagination, Spin} from 'antd';
import {PaginationProps} from 'antd/lib/pagination';


const defaultProps = {
    gutter: 20,
    span: 6,
    loading: false,
    collectionItems: null
};

export interface CollectionViewProps<T> {
    collectionItems: Array<T>,
    getKey:(item: T) => string,
    gutter?: number,
    span?: number,
    loading?: boolean,
    pagination?: PaginationProps,
}


function getOrElse<T>(t: T | undefined, _default: T): T {
    return t ? t : _default;
}

function CollectionView<T>(props: CollectionViewProps<T>) {

    const {getKey, collectionItems} = props;
    const gutter = getOrElse(props.gutter, defaultProps.gutter);
    const loading = getOrElse(props.loading, defaultProps.loading);
    const span = getOrElse(props.span, defaultProps.span);
    const pagination = getOrElse(props.pagination, {total: collectionItems.length});

    return loading ? (<Row style={{height: '350px'}}><Row className="mcs-aligner"><Spin/></Row></Row>) : (
        <Row className="mcs-table-card">
            <Row gutter={gutter}>
                {collectionItems.map(item => {
                    return (
                        <Col key={getKey(item)} span={span}>
                            {item}
                        </Col>
                    );
                })}

            </Row>
            <Row className="text-right">
                <Pagination {...pagination} />
            </Row>
        </Row>
    );
}

export default CollectionView;
