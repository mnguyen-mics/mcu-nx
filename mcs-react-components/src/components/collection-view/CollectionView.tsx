import * as React from 'react'
import { Row, Col, Spin } from 'antd'
import cuid from 'cuid'
import Pagination, { PaginationProps } from 'antd/lib/pagination/Pagination'

export interface CollectionViewProps {
	collectionItems: JSX.Element[]
	pagination?: PaginationProps
	gutter?: number
	span?: number
	loading?: boolean
}

class CollectionView extends React.Component<CollectionViewProps> {
	static defaultProps: Partial<CollectionViewProps> = {
		gutter: 20,
		span: 6,
		loading: false,
	}

	render() {
		const prefixCls = 'mcs-collection-view'
		const { collectionItems, gutter, span, pagination, loading } = this.props

		return loading ? (
			<Row style={{ height: '350px' }}>
				<Row className={`${prefixCls}-loading`}>
					<Spin />
				</Row>
			</Row>
		) : (
			<Row className={prefixCls}>
				<Row gutter={gutter}>
					{collectionItems.map((item) => {
						return (
							<Col key={item.key || cuid()} span={span!}>
								{item}
							</Col>
						)
					})}
				</Row>
				{pagination && (
					<Row className={`${prefixCls}-text-right`}>
						<Pagination {...pagination} />
					</Row>
				)}
			</Row>
		)
	}
}

export default CollectionView
