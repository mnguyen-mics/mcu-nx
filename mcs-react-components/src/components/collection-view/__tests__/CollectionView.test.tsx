import 'jest'
import * as React from 'react'
import CollectionView, { CollectionViewProps } from '../CollectionView'
import * as TestRenderer from 'react-test-renderer'
import { Col } from 'antd'

it('should display a loading collection view', () => {
	const items = []

	for (let i = 1; i <= 30; i++) {
		items.push(
			<Col key='test'>
				<li>test {i}</li>
			</Col>
		)
	}

	const props: CollectionViewProps = {
		loading: true,
		collectionItems: items,
	}

	const component = TestRenderer.create(<CollectionView {...props} />)
	const res = component.toJSON()
	expect(res).toMatchSnapshot()
})

it('should display a collection view with items and pagination', () => {
	const items = []

	for (let i = 1; i <= 30; i++) {
		items.push(
			<Col key='test'>
				<li>test {i}</li>
			</Col>
		)
	}

	const props: CollectionViewProps = {
		collectionItems: items,
		loading: false,
		gutter: 50,
		span: 30,
		pagination: {
			current: 0,
			total: 60,
		},
	}

	const component = TestRenderer.create(<CollectionView {...props} />)
	const res = component.toJSON()
	expect(res).toMatchSnapshot()
})
