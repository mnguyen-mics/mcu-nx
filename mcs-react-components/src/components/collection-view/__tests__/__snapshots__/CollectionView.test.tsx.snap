// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a collection view with items and pagination 1`] = `
<div
  className="ant-row mcs-collection-view"
>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -25,
        "marginRight": -25,
      }
    }
  >
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          1
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          2
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          3
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          4
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          5
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          6
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          7
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          8
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          9
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          10
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          11
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          12
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          13
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          14
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          15
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          16
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          17
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          18
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          19
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          20
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          21
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          22
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          23
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          24
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          25
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          26
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          27
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          28
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          29
        </li>
      </div>
    </div>
    <div
      className="ant-col-30"
      style={
        Object {
          "paddingLeft": 25,
          "paddingRight": 25,
        }
      }
    >
      <div
        className=""
        style={
          Object {
            "paddingLeft": 25,
            "paddingRight": 25,
          }
        }
      >
        <li>
          test 
          30
        </li>
      </div>
    </div>
  </div>
  <div
    className="ant-row mcs-collection-view-text-right"
  >
    <ul
      className="ant-pagination "
      style={Object {}}
      unselectable="unselectable"
    >
      <li
        aria-disabled={true}
        className="ant-pagination-disabled ant-pagination-prev"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Previous Page"
      >
        <a
          className="ant-pagination-item-link"
        >
          <i
            aria-label="icon: left"
            className="anticon anticon-left"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 0 0 0 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </i>
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-1"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={1}
      >
        <a>
          1
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-2"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={2}
      >
        <a>
          2
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-3"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={3}
      >
        <a>
          3
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-4"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={4}
      >
        <a>
          4
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-5"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={5}
      >
        <a>
          5
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-6"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={6}
      >
        <a>
          6
        </a>
      </li>
      <li
        aria-disabled={false}
        className=" ant-pagination-next"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
        title="Next Page"
      >
        <a
          className="ant-pagination-item-link"
        >
          <i
            aria-label="icon: right"
            className="anticon anticon-right"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 0 0 302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 0 0 0-50.4z"
              />
            </svg>
          </i>
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`should display a loading collection view 1`] = `
<div
  className="ant-row mcs-collection-view-loading"
>
  <div
    className="ant-row mcs-collection-view-loading-content"
  >
    <div
      className="ant-spin ant-spin-spinning"
    >
      <span
        className="ant-spin-dot ant-spin-dot-spin"
      >
        <i />
        <i />
        <i />
        <i />
      </span>
    </div>
  </div>
</div>
`;
